src/Util/Search.php

```diff
@@ -890,9 +893,12 @@ private function doesUdiExistInQueryTerm(string $queryTerm, Query\BoolQuery $fie
         if (preg_match_all($udiRegEx, $queryTerm, $matches)) {
             trim(preg_replace($udiRegEx, '', $queryTerm));
             $queryTerm = $matches[1][0];
-            // Replacing the 11th position to ":"
-            $queryTerm = substr_replace($queryTerm, ':', 11, 1);
-            $fieldsBoolQuery->addShould($this->getUdiQuery($queryTerm));
+            foreach ($matches[1] as $udi) {
+                $queryTerm = $queryTerm . ' ' . $udi;
+                // Replacing the 11th position to ":"
+                $udiQuery = substr_replace($udi, ':', 11, 1);
```

Comment:
I would use a `preg_replace` here, to find the pattern to fix, and fix it globally.